src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java

```diff
@@ -973,28 +974,19 @@ public String getCastScriptStatement(String name, String castType, List<KVValue>
      * approach will return type of result column as DOUBLE, although there is enough information to understand that
      * it might be safely treated as INTEGER.
      */
-    public static Schema.Type getScriptFunctionReturnType(Field field) {
+    public static Schema.Type getScriptFunctionReturnType(
+            int fieldIndex, Field field, ColumnTypeProvider scriptColumnType) {
+        Schema.Type returnType = null;
+        if (scriptColumnType != null && fieldIndex != -1) {
```

Comment:
In which case `fieldIndex` is -1?

Reply:
if `fieldName` is not found in the `fieldNameList`

Reply:
> if `fieldName` is not found in the `fieldNameList`

Is this possible?

Reply:
Looks like it's not- think its safe to remove based on `for (String fieldName : fieldNameList)` in `SelectResultSet`